Source/WebCore/workers/service/ServiceWorkerContainer.cpp

```diff
@@ -761,4 +765,37 @@ void ServiceWorkerContainer::getNavigationPreloadState(ServiceWorkerRegistration
     });
 }
 
+void ServiceWorkerContainer::addCookieChangeSubscriptions(ServiceWorkerRegistrationIdentifier identifier, Vector<CookieChangeSubscription>&& subscriptions, Ref<DeferredPromise>&& promise)
+{
+    ensureSWClientConnection().addCookieChangeSubscriptions(identifier, WTFMove(subscriptions), [promise = WTFMove(promise)](auto&& result) mutable {
```

Comment:
We need a smart pointer to the SWClientConnection on the stack before calling addCookieChangeSubscriptions() on it. I suggest introducing a new `ensureProtectedSWClientConnection()` which returns a `Ref< SWClientConnection >` and call it here.